### ISSUE TITLE
Request for rule avoid_unused_function_parameters

### ISSUE THREAD

**Issue by apaatsio** [dart-lang/linter#1466]
I would like to have a rule `avoid_unused_function_parameters` (similar to `avoid_unused_constructor_parameters`)

**Examples**

BAD (`sideB` is never used):
```dart
double squareArea(double sideA, double sideB) => sideA * sideA;
```
GOOD:
```dart
double squareArea(double side) => side * side;
```

**Overridden and anonymous functions**
This rule shouldn't apply to overridden or anonymous functions.

GOOD:
```dart
@override
Widget build(BuildContext context) => Text('Hello');
```
GOOD:
```dart
final ones = Iterable.generate(10, (int index) => 1);
```

**Comment by srawlins** [comment]
Considering the stipulation that this check just wouldn't apply to overriding methods, I think this could be an analyzer Hint. I can't think of a false positive.

**Comment by bwilkerson** [comment]
Sometimes people will add temporarily unused parameters for the purpose of forward compatibility.
